### PR TITLE
YAF-5286 adapt common-chart for better suport for secrets,

### PR DESCRIPTION
**Pull request by guneriu**
* list of static secrets can be passed
* introduced environmentSecrets parameter
* updated README.md for upgrade and new parameters
* updated chart release action to accept release-notes

# Pull Request

> Tracked in JIRA by [YAF-5286](https://yiluts.atlassian.net/browse/YAF-5286)

## Scope of the PR

Describe the scope of this pull request.

## Considerations / Implementation Details

Describe any considerations or implementation details.

[YAF-5286]: https://yiluts.atlassian.net/browse/YAF-5286?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: yilu-common/values-test.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Inject secrets
+secrets:
```
Comment (guneriu) [inline yilu-common/values-test.yaml:2]: these resulted with 

```
# Source: yilu-common/templates/vaultSecret.yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultStaticSecret
metadata:
  name: booking-secrets
  labels:
    simpletrip: demo-service
spec:
  namespace: admin/yiluhub/
  vaultAuthRef: vault-secrets-operator/default
  mount: kv/services/secrets
  type: kv-v2
  path: bookings
  version: 1
  refreshAfter: 1h
  destination:
    name: booking-secrets
    create: true
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultStaticSecret
metadata:
  name: payment-secrets
  labels:
    simpletrip: demo-service
spec:
  namespace: admin/yiluhub/
  vaultAuthRef: vault-secrets-operator/default
  mount: kv/services/secrets
  type: kv-v2
  path: payments
  version: 1
  refreshAfter: 1h
  destination:
    name: payment-secrets
    create: true
 ```

File: yilu-common/README.md [hunk 2]
```diff
@@ -297,7 +297,9 @@ will generate the code below, please configure your secret accordingly to match
 We have enabled Kubernetes & Vault secret integration in chart version 0.4.0 and above.
 More details can be found [here](https://yiluts.atlassian.net/wiki/spaces/YILU/pages/2463694899/HCP+Vault+-+Kubernetes+Integration)
 
-When you enable secrets, `secrets.enabled`, Kubernetes will fetch the secret from Vault. Thus make sure that corresponding secret exists in Vault. More details around adding/ updating secrets in Vault can be found [here](https://github.com/yiluhub/vault-service-secret)
+You can fetch secrets from Vault via Vault Secrets Operator(VOS).
```
Comment (davidmukiibi-yilu) [inline yilu-common/README.md:300]: Vault Secrets Operator(VOS) should be Vault Secrets Operator(VSO)

File: yilu-common/values-test.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        mountPath: "kv/services/secrets"
+        refreshInterval: 1h
+        version: ~ # set it to null to always return latest version
+        type: kv-v2
+        secretKeys:
+          - username
+          - password
+
+
+environmentSecrets:
```
Comment (guneriu) [inline yilu-common/values-test.yaml:26]: and this
```
 env:
        - name: SPRING_ENVIRONMENT
          valueFrom:
            configMapKeyRef:
              name: spring-boot-cloud
              key: spring.environment
        - name: CLOUD_ENVIRONMENT
          valueFrom:
            configMapKeyRef:
              name: spring-boot-cloud
              key: cloud.environment
        - name: SERVER_PORT
          value: "8080"## helm range changes scope.
        - name: AUTHORIZATION_TOKEN
          valueFrom:
            secretKeyRef:
              name: booking-secrets
              key: AUTHORIZATION_TOKEN
        - name: WORLDSHOP_PRICING_API
          valueFrom:
            secretKeyRef:
              name: booking-secrets
              key: WORLDSHOP_PRICING_API## helm range changes scope.
        - name: username
          valueFrom:
            secretKeyRef:
              name: payment-secrets
              key: username
        - name: password
          valueFrom:
            secretKeyRef:
              name: payment-secrets
              key: password
      
              ```

**Review by davidmukiibi-yilu — CHANGES_REQUESTED** [review]

File: yilu-common/README.md (hunk 2 above)
Comment (davidmukiibi-yilu) [reply yilu-common/README.md:300]: only change is VSO

File: yilu-common/values-test.yaml (hunk 1 above)
Comment (davidmukiibi-yilu) [reply yilu-common/values-test.yaml:2]: I see a problem here @guneriu , the 2 secrets are not separated by the three dashes... this will error out

Comment (guneriu) [reply yilu-common/values-test.yaml:2]: nice catch @davidmukiibi-yilu I'll adjust it, I was also debating if I can use  but for now it's enough https://developer.hashicorp.com/vault/docs/platform/k8s/vso/api-reference#vaultstaticsecretlist

**Review by mayank-aggarwal-yilu — DISMISSED** [review]
lgtm

File: yilu-common/templates/_helpers.tpl [hunk 4]
```diff
@@ -28,3 +28,41 @@ Usage:
 {{- .Values.image.repository -}}
 {{- end -}}
 {{- end -}}
+
```
Comment (guneriu) [inline yilu-common/templates/_helpers.tpl:31]: example outputs from tests with not valid parameters

```
helm.go:84: [debug] execution error at (yilu-common/templates/notes.txt:1:4):
VALUES VALIDATION:
common: secrets.staticSecrets
    `secretKeys` and `transformation` can't be used together
```


```
helm.go:84: [debug] execution error at (yilu-common/templates/notes.txt:1:4):
VALUES VALIDATION:
common: secrets.dynamicSecrets.aws
    `dynamicSecrets` with `aws` type and `aws` block can't be used together
```

**Comment by davidmukiibi-yilu** [comment]
do we still need the aws block?

**Review by davidmukiibi-yilu — APPROVED** [review]

File: yilu-common/values-test.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# Inject secrets
+secrets:
+  dynamicSecrets:
```
Comment (guneriu) [inline yilu-common/values-test.yaml:3]: based on the test values, template rendering results to this.

```
---
# Source: yilu-common/templates/vaultSecret.yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultDynamicSecret
metadata:
  name: hede
  labels:
    simpletrip: demo-service
spec:
  namespace: admin/yiluhub/
  vaultAuthRef: vault-secrets-operator/default
  mount: aws/secret-engine
  path:
  renewalPercent: 70
  destination:
    name: hede
    create: true
  rolloutRestartTargets:
    - kind: Deployment
      name: demo-service
---
# Source: yilu-common/templates/vaultSecret.yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultStaticSecret
metadata:
  name: booking-secrets
  labels:
    simpletrip: demo-service
spec:
  namespace: admin/yiluhub/
  vaultAuthRef: vault-secrets-operator/default
  mount: kv/services/secrets
  type: kv-v2
  path: bookings
  version: 1
  refreshAfter: 1h
  destination:
    name: booking-secrets
    create: true
---
# Source: yilu-common/templates/vaultSecret.yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultStaticSecret
metadata:
  name: payment-secrets
  labels:
    simpletrip: demo-service
spec:
  namespace: admin/yiluhub/
  vaultAuthRef: vault-secrets-operator/default
  mount: kv/services/secrets
  type: kv-v2
  path: payments
  version: 1
  refreshAfter: 1h
  destination:
    name: payment-secrets
    create: true
    transformation:
      templates:
        application.yaml:
          text: |
            {{- printf "%s" (b64dec (get .Secrets "application.yaml")) -}}
 ```

**Comment by guneriu** [comment]
> do we still need the aws block?

this will be tackled later after going through the services, if we see we don't need the feature, then we can remove this.
but I feel it can be useful, let's see.

**Pull request merged**